fastq_filtrator.py

```diff
@@ -0,0 +1,96 @@
+
+def fun_gc_bounds(gc_bounds, GC_content):
+    left_border = 0.0
+    rigth_border = 100.0
+    if isinstance(gc_bounds, float) | isinstance(gc_bounds, int) == True:
+        rigth_border = float(gc_bounds)
+    else:
+        left_border = float(gc_bounds[0])
+        rigth_border = float(gc_bounds[1])
+    if left_border < GC_content < rigth_border:
+        return(True)
+    else:
+        return(False)
+
+
+def fun_length_bounds(length_bounds, counter):
+    left_border = 0.0
+    rigth_border = float(2**32)
+    if isinstance(length_bounds, float) | isinstance(length_bounds, int) == True:
+        rigth_border = float(length_bounds)
+    else:
+        left_border = float(length_bounds[0])
+        rigth_border = float(length_bounds[1])
+    if left_border < counter < rigth_border:
+        return(True)
+    else:
+        return(False)
+
+
+def fun_qual_thresh(qual_strok, quality_threshold = 0.0):
+    qual = 0
+    cou = 0
+    for k in qual_strok:
+        qual += ord(k)
+        cou += 1
+    if (qual/cou) < quality_threshold:
+        return(False)
+    else:
+        return(True)
+
+
+
+passed = "_passed.fastq"
+
+
+def main(input_fastq, output_file_prefix, gc_bounds = (0, 100), length_bounds = (0, 2**32), quality_threshold = 0, save_filtered = False):
+    with open(input_fastq, "r") as file, open(f"{output_file_prefix}{passed}", "w") as ouf:
+        spis = []
+        for line in file:
+            spis += [line.strip()]
+        indicess = []
+        index_fail = []
+        for i in range(len(spis)):
+            if (i - 1) % 4 == 0:
+                counter = 0
+                GC = 0
+                AT = 0
+                for j in spis[i]:
+                    counter += 1
+                    if j == "G":
+                        GC += 1
+                    elif j == "C":
+                        GC += 1
+                    elif j == "A":
+                        AT += 1
+                    elif j == "T":
+                        AT += 1
+                GC_content = GC / (GC + AT) * 100
+                GC_cont = fun_gc_bounds(gc_bounds, GC_content)
+                length_OK = fun_length_bounds(length_bounds, counter)
+                if GC_cont == True and length_OK == True:
+                    indicess += [i]

```

Comment:
Можно ещё использовать `append`
```suggestion
                    indicess.append(i)

```
И ещё `+=` для списков работает довольно необычно и по-сути эквивалентно `append`, поэтому можно записать вот так
```suggestion
                    indicess += i
```
Но так делать не стоит, это просто к сведению